OpenKh.Patcher/PatcherProcessor.cs

```diff
@@ -279,5 +284,124 @@ private static void PatchAreaDataScript(Context context, List<AssetFile> sources
 
             Kh2.Ard.AreaDataScript.Write(stream.SetPosition(0), scripts.Values);
         }
+
+        private static readonly Dictionary<string, byte> characterMap = new Dictionary<string, byte>(){
+            { "Sora", 1 }, { "Donald", 2 }, { "Goofy", 3 },  { "Mickey", 4 },  { "Auron", 5 }, { "PingMulan",6 }, { "Aladdin", 7 },  { "Sparrow", 8 }, { "Beast", 9 },  { "Jack", 10 },  { "Simba", 11 }, { "Tron", 12 }, { "Riku", 13 }, { "Roxas", 14} 
+        };
+
+        private static readonly IDeserializer deserializer = new DeserializerBuilder().IgnoreUnmatchedProperties().Build();
+            
+        private static void PatchListReplace(Context context, List<AssetFile> sources, Stream stream)
+        {
+            foreach (var source in sources)
+            {
+                    string sourceText = File.ReadAllText(context.GetSourceModAssetPath(source.Name));
+                    switch (source.Type)
+                    {
+                        case "trsr":
+                            var trsrList = Kh2.SystemData.Trsr.Read(stream).ToDictionary(x => x.Id, x => x);
+                            var moddedTrsr = deserializer.Deserialize<Dictionary<int, Kh2.SystemData.Trsr>>(sourceText);
+                            foreach (var treasure in moddedTrsr)
+                            {
+                                trsrList[treasure.Value.Id].ItemId = treasure.Value.ItemId;
```

Comment:
I can see what you are doing here and why. But if in the future someone wants to effectively patch the `trsr` with the other allowed values, modifying this implementation to support it will break the existing mods. What if you check the world value? If it is `0` then patch the `ItemId`. If it is not `0` then patch everything in the record.

Reply:
Also what if the Value ID (or the index of the array) is too big? You might want to add a new item to the list at that point.